bucket/eiskaltdcpp.json

```
@@ -0,0 +1,38 @@
+{
+    "version": "2.4.2",
+    "description": "File sharing program using DC and ADC protocols",
+    "homepage": "https://github.com/eiskaltdcpp/eiskaltdcpp",
+    "license": "GPL-3",
+    "architecture": {
+        "64bit": {
+            "url": "https://downloads.sourceforge.net/project/eiskaltdcpp/files/Windows/EiskaltDC++-2.4.2_x86_64-portable.7z",
+            "hash": "sha1:9ed0cd684989e40a29d58e85960f908d780774c0"
+        },
+        "32bit": {
+            "url": "https://downloads.sourceforge.net/project/eiskaltdcpp/files/Windows/EiskaltDC++-2.4.2_x86-portable.7z",
+            "hash": "sha1:764df76901b038d460909f4422950082c70c6869"
+        }
+    },
+    "extract_dir": "EiskaltDC++-2.4.2_x86_64",
+    "bin": "eiskaltdcpp-daemon.exe",
```

```suggestion
    "pre_install": [
        "'FinishedDownloads.sqlite', 'FinishedUploads.sqlite', 'HashData.dat' | ForEach-Object {",
        "    if(!(Test-Path \"$persist_dir\\$_\")) {New-Item \"$dir\\$_\" | Out-Null}",
        "}"
    ],
    "bin": "eiskaltdcpp-daemon.exe",
```
This is to let **persist** work as we want. Since `FinishedDownloads.sqlite` and other files does not exist in the archive (but rather created in the first run of `EiskaltDC++.exe`), we need pre_install to make it work properly.

---

Is there any scoop package manifest generator? I use https://n8225.github.io for awesome-selfhosted list BTW

---

> Is there any scoop package manifest generator?

I don't use manifest generators, so I cannot give you information about that. Sorry.
You can check [our wiki](https://github.com/ScoopInstaller/Scoop/wiki/App-Manifests) for details about the fields.

By the way, we need `pre_install` for files because when the **persist file** does not exist after extracting the archive, *Scoop* automatically assumes it as a **folder**, therefore we need to create the file first.